Source/Pawnmorphs/Esoteria/PawnTransferUtilities.cs

```diff
@@ -269,11 +301,69 @@ public static void TransferFavor([NotNull] Pawn pawn1, [NotNull] Pawn pawn2)
                 if (otherRecord == null && hediff.Part != null)
                     continue;
 
-                if (health2.hediffSet.HasHediff(hediff.def, otherRecord)) 
-                    continue;
-
+                if (health2.hediffSet.HasHediff(hediff.def, otherRecord))
+                { 
+                    Hediff otherHediff = health2.hediffSet.GetFirstHediffOfDef(hediff.def);
+                    if (hediff.Severity == otherHediff.Severity)
+                    {
+                        continue;
+                    }
+                    else
+                    {
+                        if (otherHediff is Hediff_Psylink psylinkExitsting)
+                        {
+                            psylinkExitsting.ChangeLevel((int)(hediff.Severity - psylinkExitsting.Severity), false);
+                        }
+                        else if (otherHediff is Hediff_Level newLevelExitsting)
+                        {
+                            newLevelExitsting.SetLevelTo((int)hediff.Severity);
+                        }
+                        else
+                        {
+                            otherHediff.Severity = hediff.Severity;
+                        }
+                    }
+                }
```

Comment:
All of this code is basically identical to what is further below.
Could merge this if you handle "Find or Make + Suppress & add" and then handle severity adjustments